### ISSUE TITLE
[Bug]:  TypeError: 'NoneType' object is not iterable when generating txt2img 

### ISSUE THREAD

**Issue by shalim2904** [lllyasviel/stable-diffusion-webui-forge#727]
### Checklist

- [X] The issue exists after disabling all extensions
- [X] The issue exists on a clean installation of webui
- [ ] The issue is caused by an extension, but I believe it is caused by a bug in the webui
- [ ] The issue exists in the current version of the webui
- [ ] The issue has not been reported before recently
- [ ] The issue has been reported before but has not been fixed yet

### What happened?

txt2img generation cause  TypeError: 'NoneType' object is not iterable it happened at latest build also on  build "b9705c5"

### Steps to reproduce the problem

1. run forge from stability matrix
2. disable all extensions
3. enter prompt in txt2img

### What should have happened?

generate image

### What browsers do you use to access the UI ?

Mozilla Firefox

### Sysinfo

[sysinfo-2024-05-12-04-25.json](https://github.com/lllyasviel/stable-diffusion-webui-forge/files/15285093/sysinfo-2024-05-12-04-25.json)

### Console logs

```Shell
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Version: f0.0.17v1.8.0rc-latest-273-gb9705c58
Commit hash: b9705c58f66c6fd2c4a0168b26c5cf1fa6c0dde3
Installing requirements
Launching Web UI with arguments: --directml --skip-torch-cuda-test --no-half-vae --gradio-allowed-path D:\APP\StabilityMatrix-win-x64\Data\Images
Using directml with device: 
Total VRAM 1024 MB, total RAM 28417 MB
Trying to enable lowvram mode because your GPU seems to have 4GB or less. If you don't want this use: --always-normal-vram
Set vram state to: LOW_VRAM
Device: privateuseone
VAE dtype: torch.float32
CUDA Stream Activated:  False
Warning: caught exception 'Torch not compiled with CUDA enabled', memory monitor disabled
Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --attention-split
==============================================================================
You are running torch 2.0.0+cpu.
The program is tested to work with torch 2.1.2.
To reinstall the desired version, run with commandline flag --reinstall-torch.
Beware that this will cause a lot of large files to be downloaded, as well as
there are reports of issues with training tab on the latest version.

Use --skip-version-check commandline argument to disable this check.
==============================================================================
Loading weights [fc2511737a] from D:\APP\StabilityMatrix-win-x64\Data\Packages\Stable Diffusion WebUI Forge\models\Stable-diffusion\chilloutmix_NiPrunedFp32Fix.safetensors
model_type EPS
UNet ADM Dimension 0
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 14.4s (prepare environment: 5.3s, import torch: 3.9s, import gradio: 1.2s, setup paths: 1.2s, initialize shared: 0.1s, other imports: 0.7s, load scripts: 0.7s, create ui: 0.6s, gradio launch: 0.4s).
Using split attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using split attention in VAE
extra {'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_l.text_projection'}
left over keys: dict_keys(['model_ema.decay', 'model_ema.num_updates'])
To load target model SD1ClipModel
Begin to load 1 model
Moving model(s) has taken 0.01 seconds
Model loaded in 3.4s (load weights from disk: 0.4s, forge load real models: 2.0s, load VAE: 0.4s, calculate empty prompt: 0.6s).
Traceback (most recent call last):
  File "D:\APP\StabilityMatrix-win-x64\Data\Packages\Stable Diffusion WebUI Forge\modules_forge\main_thread.py", line 37, in loop
    task.work()
  File "D:\APP\StabilityMatrix-win-x64\Data\Packages\Stable Diffusion WebUI Forge\modules_forge\main_thread.py", line 26, in work
    self.result = self.func(*self.args, **self.kwargs)
  File "D:\APP\StabilityMatrix-win-x64\Data\Packages\Stable Diffusion WebUI Forge\modules\txt2img.py", line 111, in txt2img_function
    processed = processing.process_images(p)
  File "D:\APP\StabilityMatrix-win-x64\Data\Packages\Stable Diffusion WebUI Forge\modules\processing.py", line 752, in process_images
    res = process_images_inner(p)
  File "D:\APP\StabilityMatrix-win-x64\Data\Packages\Stable Diffusion WebUI Forge\modules\processing.py", line 848, in process_images_inner
    p.rng = rng.ImageRNG((opt_C, p.height // opt_f, p.width // opt_f), p.seeds, subseeds=p.subseeds, subseed_strength=p.subseed_strength, seed_resize_from_h=p.seed_resize_from_h, seed_resize_from_w=p.seed_resize_from_w)
  File "D:\APP\StabilityMatrix-win-x64\Data\Packages\Stable Diffusion WebUI Forge\modules\rng.py", line 114, in __init__
    self.generators = [create_generator(seed) for seed in seeds]
  File "D:\APP\StabilityMatrix-win-x64\Data\Packages\Stable Diffusion WebUI Forge\modules\rng.py", line 114, in <listcomp>
    self.generators = [create_generator(seed) for seed in seeds]
  File "D:\APP\StabilityMatrix-win-x64\Data\Packages\Stable Diffusion WebUI Forge\modules\rng.py", line 86, in create_generator
    generator = torch.Generator(device).manual_seed(int(seed))
RuntimeError: Device type privateuseone is not supported for torch.Generator() api.
Device type privateuseone is not supported for torch.Generator() api.
*** Error completing request
*** Arguments: ('task(vox45t86iw72py1)', <gradio.routes.Request object at 0x00000210CEC9F280>, 'girl in meadow', '', [], 20, 'DPM++ 2M Karras', 1, 1, 7, 512, 512, False, 0.7, 2, 'Latent', 0, 0, 0, 'Use same checkpoint', 'Use same sampler', '', '', [], 0, False, '', 0.8, -1, False, -1, 0, 0, 0, False, False, 'positive', 'comma', 0, False, False, 'start', '', 1, '', [], 0, '', [], 0, '', [], True, False, False, False, False, False, False, 0, False) {}
    Traceback (most recent call last):
      File "D:\APP\StabilityMatrix-win-x64\Data\Packages\Stable Diffusion WebUI Forge\modules\call_queue.py", line 57, in f
        res = list(func(*args, **kwargs))
    TypeError: 'NoneType' object is not iterable

---
```

**Comment by LadyFlames** [comment]
your not the only one with the same issue its happening to me as well and yes i can confirm that with or without using extensions/Integrated extensions it still happens and Going to Different Commit and Switching Branches doesnt help the issue is Still there

**Comment by LadyFlames** [comment]
i got it to work just Delete all External Extentions and dont use any of the integrated extensions what i mean is that once you launch the UI dont Activate any of integrated Extensions and dont use Any Scripts either if you do it that way it works Normally Without Any Issues at least it works for me when i did that

**Comment by LadyFlames** [comment]
Just Tried again with the Extension and Scripts turned on and it didnt Work at all This happened   File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules_forge\main_thread.py", line 37, in loop
    task.work()
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules_forge\main_thread.py", line 26, in work
    self.result = self.func(*self.args, **self.kwargs)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules\txt2img.py", line 108, in txt2img_function
    processed = modules.scripts.scripts_txt2img.run(p, *p.script_args)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules\scripts.py", line 785, in run
    processed = script.run(p, *script_args)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\scripts\AutoChar.py", line 850, in run
    mid_image_face_inpaint = inpaint(last_image_batch, mask_face, inpaint_face_size[0],
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\scripts\AutoChar.py", line 809, in inpaint
    inpaint_output = process_images(instance_inpaint)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules\processing.py", line 752, in process_images
    res = process_images_inner(p)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules\processing.py", line 922, in process_images_inner
    samples_ddim = p.sample(conditioning=p.c, unconditional_conditioning=p.uc, seeds=p.seeds, subseeds=p.subseeds, subseed_strength=p.subseed_strength, prompts=p.prompts)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules\processing.py", line 1703, in sample
    samples = self.sampler.sample_img2img(self, self.init_latent, x, conditioning, unconditional_conditioning, image_conditioning=self.image_conditioning)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules\sd_samplers_timesteps.py", line 142, in sample_img2img
    samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules\sd_samplers_common.py", line 263, in launch_sampling
    return func()
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules\sd_samplers_timesteps.py", line 142, in <lambda>
    samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules\sd_samplers_timesteps_impl.py", line 24, in ddim
    e_t = model(x, timesteps[index].item() * s_in, **extra_args)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules\sd_samplers_cfg_denoiser.py", line 182, in forward
    denoised = forge_sampler.forge_sample(self, denoiser_params=denoiser_params,
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\modules_forge\forge_sampler.py", line 88, in forge_sample
    denoised = sampling_function(model, x, timestep, uncond, cond, cond_scale, model_options, seed)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\ldm_patched\modules\samplers.py", line 289, in sampling_function
    cond_pred, uncond_pred = calc_cond_uncond_batch(model, cond, uncond_, x, timestep, model_options)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\ldm_patched\modules\samplers.py", line 258, in calc_cond_uncond_batch
    output = model.apply_model(input_x, timestep_, **c).chunk(batch_chunks)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\ldm_patched\modules\model_base.py", line 90, in apply_model
    model_output = self.diffusion_model(xc, t, context=context, control=control, transformer_options=transformer_options, **extra_conds).float()
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\ldm_patched\ldm\modules\diffusionmodules\openaimodel.py", line 867, in forward
    h = forward_timestep_embed(module, h, emb, context, transformer_options, time_context=time_context, num_video_frames=num_video_frames, image_only_indicator=image_only_indicator)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\ldm_patched\ldm\modules\diffusionmodules\openaimodel.py", line 61, in forward_timestep_embed
    x = layer(x)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\ldm_patched\modules\ops.py", line 114, in forward
    return super().forward(*args, **kwargs)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\nn\modules\conv.py", line 460, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "D:\Stability Matrix\Packages\Stable Diffusion WebUI Forge\venv\lib\site-packages\torch\nn\modules\conv.py", line 456, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: Given groups=1, weight of size [320, 9, 3, 3], expected input[1, 4, 118, 99] to have 9 channels, but got 4 channels instead
Given groups=1, weight of size [320, 9, 3, 3], expected input[1, 4, 118, 99] to have 9 channels, but got 4 channels instead
*** Error completing request
*** Arguments: ('task(p7tla62hyi3t8si)', <gradio.routes.Request object at 0x0000026E05AF9F90>  TypeError: 'NoneType' object is not iterable

**Comment by georgitsenov** [comment]
I am getting this in txt2img as well:
TypeError: 'NoneType' object is not iterable

I figured it comes from torch difference. Your install script installs torch==2.1.2+cu121
I tried upgrading torch==2.3.0 and the issue is no longer present for me.

**Comment by LastTargaryen** [comment]
@georgitsenov , This has happened to me and I'm using latest torch+cu121--To Everyone: this always fixes it for me:

When I get that error, I change checkpoints, click generate, it works, then go back to original checkpoint and it works.
*maybe important??: I'm almost always using sd1.5 checkpoints, don't know if that has anything to do with the viability of this workaround.

I got really desperate once when I got the error, cuz up until that point, the error meant you had to reload the UI, which I hate lol so I turned off each extension one-by-one and what ended up doing it was changing checkpoints. Since then, when I get the error, I do the checkpoint switch and it has worked every single time. 

hope it helps
